views/forms/BarChartFields.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from 'react'
+import Delete from '@material-ui/icons/Delete';
+import TextField from '@material-ui/core/TextField'
+import Button from '@material-ui/core/Button'
+import Typography from '@material-ui/core/Typography';
+
+const BarChartFields = ({ 
+  classes,
+  updateFormValues,
+  formValues,
+  setFormValues,
+}) => {
+  const { title } = formValues
+
+  const [chartTitle, setChartTitle] = useState('')
+  
+  const addValueAndLabelField = () => setFormValues(prevState => ({
+    ...prevState,
+    fieldLabelValueMap: [...prevState
+      .fieldLabelValueMap,
+        { 
+          value: '', 
+          label: '' 
+        }
+      ]
+    })
+  )
+  const removeValueAndLabelField = (id) => setFormValues((prevState) => ({ 
+    ...prevState,
+    fieldLabelValueMap: [...prevState
+      .fieldLabelValueMap
+      .filter((_, index) => index !== id)
+      ]
+    })
+  )
+  const handleValueAndLabelFieldUpdate = (e, id) => setFormValues((prevState) => ({
+    ...prevState, 
+    fieldLabelValueMap: [...prevState
+      .fieldLabelValueMap
+        .map((field, idx) => 
+          id === idx
+          ? 
+            ({ ...field, [e.target.name]: e.target.value }) 
+          : 
+            ({ ...field })
+          )
+        ]
+      })
+    )
```

Comment:
Minor formatting suggestions to make this a bit easier to read.

```suggestion
  const handleValueAndLabelFieldUpdate = (e, id) => setFormValues((prevState) => ({
    ...prevState, 
    fieldLabelValueMap: [
      ...prevState.fieldLabelValueMap.map((field, idx) => 
        id === idx
          ? { ...field, [e.target.name]: e.target.value }
          : field
      )
    ],
  }))
```

Reply:
[ab607bc](https://github.com/AMP-SCZ/dpdash/pull/84/commits/ab607bcdc9895d5c9da30b9ab50689aab6b3b00e)